src/test/java/com/jcabi/xml/XMLDocumentTest.java

```diff
@@ -374,4 +375,32 @@ public void preservesXmlNamespaces() throws Exception {
         );
     }
 
+    /**
+     * XMLDocument can generate a compact representation of itself.
+     * @throws Exception If something goes wrong inside
+     */
+    @Test
+    public void compacts() throws Exception {
+        final XML xml = new XMLDocument(
+            "<a> \n\t  <b/>\n <c> text </c> \n\n\t </a>"
+        );
+        final XML compact = xml.compact();
+        MatcherAssert.assertThat(
+            compact,
+            Matchers.not(
+                Matchers.equalTo(
+                    XML.class.cast(
+                        new XMLDocument("<a><b/>xyz<c> text </c></a>")
```

Comment:
@krzyk trying to make sure `compact()` does what it is supposed to by making sure it is equal to what it is supposed to be and not equal to something different. Don't you think that is related?